### PR TITLE
feat: add encrypt and decrypt methods for signing + TypeScript upgrade

### PR DESCRIPTION
**Pull request by karrui**
## Problem
Some of our future services may include signing fields before encrypting the data, and this PR aims to add functionality to our current methods.

## BREAKING Changes:
Note the change in ordering of the function signature and the new optional parameter `signingSecretKey`
```diff
- formsg.crypto.encrypt(encryptionPublicKey, publicKey, message) : string
+ formsg.crypto.encrypt(message, encryptionPublicKey, signingSecretKey) : string
```
The return type of the decrypt method in the sdk has also changed from just returning the decrypted array of responses:
```diff
- formsg.crypto.decrypt(encryptionSecretKey: string, encryptedContent: string) : FormField[] | null
+ formsg.crypto.decrypt(encryptionSecretKey: string, encryptedContent: string, verifiedContent?: string) 
+   : {
+      responses: FormField[]
+      verified?: Record<string, any>
+   } | null
```

The whole package has also been rewritten in TypeScript for better typing and maintainability.

### CODE REVIEW

File: spec/crypto.spec.ts [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+
+    // Assert
+    expect(decrypted).toHaveProperty('responses', plaintext)
+  })
+
+  it('should be able to encrypt and sign submissions if signingPrivateKey is given', () => {
+    // Arrange
+    const { publicKey, secretKey } = formsg.crypto.generate()
+    const mockVerifiedContent = {
+      question: 'SingPass Validated NRIC',
```
Comment (liangyuanruo) [inline spec/crypto.spec.ts:90]: suggest updating the format of this mock verified content to what's actually being used

File: src/util/guard.ts [hunk 2]
```diff
@@ -0,0 +1,22 @@
+function determineIsFormFields(tbd: any): tbd is FormField[] {
```
Comment (liangyuanruo) [inline src/util/guard.ts:1]: data validation has a lot of potential for bugs and should be sharded out to a library such as [Joi](https://hapi.dev/module/joi/#introduction) or validator.js.

also suggest renaming the file/function to say, `validate.ts` and `validateFormFields`

**Comment by liangyuanruo** [comment]
OK, I've checked and ES5 is probably safe with IE11. Please test with IE11 to confirm. Please remove `.babelrc` file if babel is no longer being used during build.

**Comment by karrui** [comment]
> I notice the typescript compiler targets ES5 but not IE11 specifically. Are there guarantees that this is sufficient for IE11 interoperability, or do we still need a pass with Babel after typescript transpile? Is babel even used at all during build?

ES5 already targets ie11, see https://kangax.github.io/compat-table/es5/

**Pull request merged**